### PR TITLE
scripts: compliance: check for binding compatible and property formats

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Add a pair of extra checks to the existing DevicetreeBindingsCheck to validate the characters used in compatible and properties value.

Ran against the existing bindings, all compatibles match, few properties are off:

https://gist.github.com/fabiobaltieri/8be56c78e1220d9e09bdb199ced42cda

### CODE REVIEW

**Comment by carlescufi** [comment]
Assigning to @mbolivar-nordic given his familiarity with this part of the code.

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -253,6 +256,38 @@ def required_false_check(self, dts_binding):
...
+    def compatible_properties(self, dts_binding):
+        with open(dts_binding) as file:
+            binding = yaml.load(file, Loader=yaml.SafeLoader)
+
+            property_names = []
+            property_names.extend(binding.get("properties", {}))
+
+            child_binding = binding.get("child-binding", None)
+            if child_binding:
+                property_names.extend(child_binding.get("properties", {}))
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:282]: note: we can have child-binding at arbitrary levels (forming e.g. grandchild binding)

Comment (fabiobaltieri) [reply scripts/ci/check_compliance.py]: Good point, let's iterate on those as well.

**Comment by fabiobaltieri** [comment]
- added recursion on multiple `child-binding`

**Review by gmarull — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Comment by cfriedt** [comment]
@fabiobaltieri - are you waiting on any particular reviewers for this?

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri - are you waiting on any particular reviewers for this?

Yeah, @mbolivar-nordic since he's the one who wrote the guideline.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
I think this should be done using edtlib instead of partially duplicating the bindings parsing logic in an external script.

**Comment by fabiobaltieri** [comment]
> I think this should be done using edtlib instead of partially duplicating the bindings parsing logic in an external script.

The problem with using `edtlib.Binding()` is that it really wants to resolve the includes, but to do that I'd have to rebuild a map of the known bindings, which implies knowing where they are. Normally DTS_ROOT_BINDINGS is set by the buildsystem, but I don't have that in the compliance script so to use that I'd have to duplicate the parsing logic here, and even then I think it may be a problem for bindings in modules. Or I could modify edtlib to have an option to ignore includes I guess, but I think at that point we may as well keep loading them directly.

**Comment by fabiobaltieri** [comment]
Woops, reused the branch for something unrelated. Anyway, looks like this is not going to go, closing in favor of Marti's version, the checks are different but will look into discussing the ones I was proposing here once that one is merged.

**Pull request closed without merging**